src/machine/CoffeeMachine.java

```diff
@@ -0,0 +1,111 @@
+package machine;
+
+import java.util.Scanner;
+
+class CoffeeMachine {
+
+    private int amountOfWater;
+    private int amountOfMilk;
+    private int amountOfCoffeeBeans;
+    private int amountOfDisposableCups;
+    private int money;
+    private CoffeeMachineState currentState;
+
+    CoffeeMachineState getCurrentState() {
+        return this.currentState;
+    }
+
+    CoffeeMachine(int amountOfWater, int amountOfMilk, int amountOfCoffeeBeans,
+                         int amountOfDisposableCups, int money, CoffeeMachineState currentState) {
+        this.amountOfWater = amountOfWater;
+        this.amountOfMilk = amountOfMilk;
+        this.amountOfCoffeeBeans = amountOfCoffeeBeans;
+        this.amountOfDisposableCups = amountOfDisposableCups;
+        this.money = money;
+        this.currentState = currentState;
+    }
+
+    private void displayContent() {
+        System.out.println("\nThe coffee machine has:");
+        System.out.printf("%d ml of water\n", this.amountOfWater);
+        System.out.printf("%d ml of milk\n", this.amountOfMilk);
+        System.out.printf("%d g of coffee beans\n", this.amountOfCoffeeBeans);
+        System.out.printf("%d disposable cups\n", this.amountOfDisposableCups);
+        System.out.printf("$%d of money\n", this.money);
+    }
+
+    private void prepareCoffee(int chosenOption) {
+        Coffee coffee = null;
+        switch (chosenOption) {
+            case 1 -> coffee = Coffee.ESPRESSO;
+            case 2 -> coffee = Coffee.LATTE;
+            case 3 -> coffee = Coffee.CAPPUCCINO;
+            default -> System.out.println("Option not found");
+        }
+        if (coffee != null) {
+            if (!isEnoughIngredients(coffee)) return;
+            this.amountOfWater -= coffee.getAmountOfWater();
+            this.amountOfMilk -= coffee.getAmountOfMilk();
+            this.amountOfCoffeeBeans -= coffee.getAmountOfCoffeeBeans();
+            this.amountOfDisposableCups -= coffee.getAmountOfDisposableCups();
+            this.money += coffee.getPrice();
+        }
+    }
+
+    private void fillSupplies() {
```

Comment:
Looks like we could add another coffee machine state `FILLING_SUPPLIES`.

Then we should get rid of this `Scanner` inside the `CoffeeMachine` and read everything inside `CoffeeMachineCommandsScanner`